src/clientdlg.cpp

```diff
@@ -254,6 +254,9 @@ CClientDlg::CClientDlg ( CClient*         pNCliP,
     // set window title (with no clients connected -> "0")
     SetMyWindowTitle ( 0 );
 
+    // track number of clients to detect joins/leaves for audio alerts
+    iClients = 0;
```

Comment:
Shouldn't this happen every time I connect to a server?

Reply:
Wouldn't this be covered by OnNumClientsChanged? As it sets the absolute value, it doesn't necessarily have to be zeroed on connect, right?

Reply:
OnNumClientsChanged tests for a change, doesn't it?  If when I was on server A iClients was 3 and I join server B - what should happen?  What if there are 3 clients there, too -- all different ones?

Reply:
> OnNumClientsChanged tests for an increase, doesn't it?

Yes, and afterwards (even if the test was false) it stores the new value.

I've added the following logging for testing locally:
```
diff --git a/src/clientdlg.cpp b/src/clientdlg.cpp
index a488365a..849564ae 100644
--- a/src/clientdlg.cpp
+++ b/src/clientdlg.cpp
@@ -885,6 +885,7 @@ void CClientDlg::OnConClientListMesReceived ( CVector<CChannelInfo> vecChanInfo
 
 void CClientDlg::OnNumClientsChanged ( int iNewNumClients )
 {
+    qInfo() << "OnNumClientsChanged iNewNumClients =" << iNewNumClients << ", iClients =" << iClients;
     if ( pSettings->bEnableAudioAlerts && iNewNumClients > iClients )
     {
         QSoundEffect* sf = new QSoundEffect();
```

When joining an empty server, this happens:
```
OnNumClientsChanged iNewNumClients = 0 , iClients = 0 <- this would additionally reset
OnNumClientsChanged iNewNumClients = 1 , iClients = 0 <- this triggers the sound
OnNumClientsChanged iNewNumClients = 1 , iClients = 1 <- this is just a duplicate and doesn't trigger the sound

OnNumClientsChanged iNewNumClients = 0 , iClients = 1 <- on disconnect, this properly resets
```

(Similar behavior on non-empty servers)

So, as far as I can see, it works properly and it can also be explained why it does. :)

This is the call chain which guarantees a proper reset on disconnect:
CClientDlg::Disconnect() -> MainMixerBoard->HideAll() -> emit NumClientsChanged ( 0 ); -> CClientDlg::OnNumClientsChanged()

Reply:
Great, thanks for checking.  I think it's worth a comment in the code (in `OnNumClientsChanged` where it stores `iNewNumClients` into `iClients` that this will reset the value to zero when connecting to a new server.